Whiskey/Tasks/Invoke-WhiskeyNUnit3Task.ps1

```diff
@@ -0,0 +1,310 @@
+    * `CoverageFilter`: a list of OpenCover filters to apply to selectively include or exclude assemblies and classes from OpenCover coverage results.
+    * `OpenCoverVersion`: the version of OpenCover to use. Defaults to the latest version available.
+    * `OpenCoverArgument`: a list of additional arguments to pass to the OpenCover console.
+    * `ReportGeneratorVersion`: the version of ReportGenerator to use. Defaults to the latest version available.
+    * `ReportGeneratorArgument`: a list of additional arguments to pass to the ReportGenerator console.
+
+    .EXAMPLE
+
+    # Example 1
+
+        BuildTasks:
+        - NUnit3:
+            Path:
+            - RootAssembly.dll
+            - subfolder\subAssembly.dll
+
+    This example will run the NUnit tests for both the `RootAssembly.dll` and the `<build root>\subfolder\subAssembly.dll` with the default .NET framework version `4.0`. OpenCover and ReportGenerator will also be run against the results of the NUnit tests.
+
+    # Example 2
+
+        BuildTasks:
+        - NUnit3:
+            Path:
+            - Assembly.dll
+            Framework: 4.5
+            DisableCodeCoverage: true
+
+    This example will run the NUnit tests for the `Assembly.dll` file using .NET framework 4.5. OpenCover and ReportGenerator will not be run after the NUnit tests have completed since `DisableCodeCoverage` was `true`.
+
+    # Example 3
+
+        BuildTasks:
+        - NUnit3:
+            Path:
+            - Assembly.dll
+            TestFilter:
+            - cat == 'Slow Data Tests' && Priority == High
+            - cat == 'Standard Tests'
+            Argument:
+            - "--debug"
+
+    This example will run the NUnit tests for the `Assembly.dll` file using the TestFilter of `(cat == 'Slow Data Tests' && Priority == High) or (cat == 'Standard Tests')` to select which tests to run. The NUnit console will be executed with the additional given argument `--debug`.
+
+    # Example 4
+
+        BuildTasks:
+        - NUnit3:
+            Path:
+            - Assembly.dll
+            CoverageFilter:
+            - "+[Open*]*"
+            - "-[Open.Test]*"
+            OpenCoverVersion: 4.6.519
+            OpenCoverArgument:
+            - "-showunvisited"
+            ReportGeneratorVersion: 2.5.11
+            ReportGeneratorArgument:
+            - "-reporttypes: Latex"
+            - "-verbosity:Verbose"
+
+    This example will run all tests located in the `Assembly.dll` file using the default .NET framework version `4.0`. OpenCover version `4.6.519` will be run against against the NUnit test results with the `CoverageFilter` `+[Open*]* -[Open.Test]*` and with the argument `-showunvisited`. ReportGenerator version `2.5.11` will be run against the OpenCover results using the arguments `-reporttypes:Latex` and `-verbosity:Verbose`.
+    #>
+
+    [CmdletBinding()]
+    [Whiskey.Task("NUnit3",SupportsClean=$true)]

```

Comment:
It should also support initialization, e.g. `SupportsInitialize=$true`. Scott just committed this change to the NUnit 2 task. Please add this functionality, along with tests.

Reply:
Fixed.